File: JavaScript Solutions/1040.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let notas = lines.shift().split(' '),
    n1 = parseFloat(notas[0]),
    n2 = parseFloat(notas[1]),
    n3 = parseFloat(notas[2]),
    n4 = parseFloat(notas[3]),
    media = ( (n1*2) + (n2*3) + (n3*4) + (n4) )/10;
    
console.log("Media: "+media.toFixed(1));
    
if (media >= 7)
    console.log("Aluno aprovado.");
    
else if (media < 5)
    console.log("Aluno reprovado.");

else {
    

    console.log("Aluno em exame.");
    
    let exame = parseFloat(lines.shift()),
        mediaFinal = (media+exame)/2.0;
    
    console.log("Nota do exame: "+exame.toFixed(1));
    if (mediaFinal < 5)
        console.log("Aluno reprovado.");
        
    else
        console.log("Aluno aprovado.");
        
    console.log("Media final: "+mediaFinal.toFixed(1));
}

File: Python Solutions/1040.py
notas= input().split()

n1= float(notas[0])
n2= float(notas[1])
n3= float(notas[2])
n4= float(notas[3])

media= ( (n1*2) + (n2*3) + (n3*4) + (n4*1) ) / 10
print(f"Media: {media:.1f}")

if media >= 7.0:
    print("Aluno aprovado.")

elif media < 5.0:
    print("Aluno reprovado.")

else:
    print("Aluno em exame.")
    n5= float(input())
    print(f"Nota do exame: {n5:.1f}")
    mediaF= (media+n5)/2
    
    if mediaF >= 5.0:
        print("Aluno aprovado.")
        
    else:
        print("Aluno reprovado.")
        
    print(f"Media final: {mediaF:.1f}")
